packages/desktop-client/src/components/reports/util.ts

```diff
@@ -13,23 +17,23 @@ export async function runAll(queries, cb) {
   cb(data);
 }
 
-export function index(data, field, mapper) {
+export function index(data, field: string, mapper?: (input) => unknown) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏could be improved with generics
```suggestion
export function index<T, K extends keyof T>(data: T[], field: K, mapper?: (input: T[K]) => string) {
```

Reply:
Please take a look at this. I implemented it using generics this time

Reply:
the generics I implemented using the way you suggested were failing the type checks for some reason with the error Argument of type 'string' is not assignable to parameter of type 'never'. I tried playing around with it but couldn't fix it. Had to finally resort to using string as the type for some fields